flink-connectors/flink-connector-kafka/src/main/java/org/apache/flink/streaming/connectors/kafka/table/KafkaOptions.java

```diff
@@ -361,18 +360,25 @@ public static void validateSourceTopic(ReadableConfig tableOptions) {
     }
 
     public static void validateSinkTopic(ReadableConfig tableOptions) {
-        String errorMessageTemp =
-                "Flink Kafka sink currently only supports single topic, but got %s: %s.";
-        if (!isSingleTopic(tableOptions)) {
-            if (tableOptions.getOptional(TOPIC_PATTERN).isPresent()) {
+        if (tableOptions.getOptional(TOPIC).isPresent()) {
+            if (tableOptions.get(TOPIC).isEmpty()) {
                 throw new ValidationException(
                         String.format(
-                                errorMessageTemp,
-                                "'topic-pattern'",
-                                tableOptions.get(TOPIC_PATTERN)));
-            } else {
-                throw new ValidationException(
-                        String.format(errorMessageTemp, "'topic'", tableOptions.get(TOPIC)));
+                                "Flink Kafka sink doesn't support empty topic, but got %s: %s.",
+                                "'topic'", tableOptions.get(TOPIC)));
```

Comment:
nit: it isn't really necessary to call `tableOptions.get(TOPIC)` here again since this will be called exactly when the list is empty.